osdk/src/cli.rs

```diff
@@ -68,10 +78,24 @@ pub enum OsdkSubcommand {
     Run(RunArgs),
     #[command(about = "Execute kernel mode unit test by starting a VMM")]
     Test(TestArgs),
-    #[command(about = "Analyze the current package and report errors")]
+    #[command(about = "Wrapper for the `cargo check` command")]
     Check,
-    #[command(about = "Check the current package and catch common mistakes")]
+    #[command(about = "Wrapper for the `cargo clippy` command")]
     Clippy,
+    #[command(about = "Wrapper for the `cargo doc` command")]
+    Doc,
```

Comment:
I think these wrapper commands should also be parsed by `clap`, instead of be parsed manually. Like the following code: 

```rust

pub enum OsdkSubcommand {
    // Other subcommands......

    #[command(about = "Analyze the current package and report errors")]
    Check(WrapperArguments),
    #[command(about = "Check the current package and catch common mistakes")]
    Clippy(WrapperArguments),
    #[command(about = "Build a package's documentation")]
    Doc(WrapperArguments)
}

#[derive(Debug, Parser)]
pub struct WrapperArgument {
    #[arg(help = "Additional Cargo arguments", allow_hyphen_values = true)]
    pub args: Vec<String>
}
```